crates/octez/src/async/baker.rs

```diff
@@ -0,0 +1,197 @@
+use crate::{Endpoint, OctezNodeConfig};
+use anyhow::{anyhow, Result};
+use std::{fmt::Display, path::PathBuf};
+use tokio::process::{Child, Command};
+
+use super::{AsyncOctezClient, Protocol};
+
+#[derive(PartialEq, Debug, Clone)]
+pub enum BakerBinaryPath {
+    BuiltIn(Protocol), // The binary exists in $PATH
+    Custom(PathBuf),   // The binary is at the given path
+}
+
+impl Display for BakerBinaryPath {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            BakerBinaryPath::BuiltIn(Protocol::Alpha) => write!(f, "octez-baker-alpha"),
+            BakerBinaryPath::BuiltIn(Protocol::ParisC) => {
+                write!(f, "octez-baker-PsParisC")
+            }
+            BakerBinaryPath::BuiltIn(Protocol::Quebec) => {
+                write!(f, "octez-baker-PsQuebec")
+            }
+            BakerBinaryPath::Custom(path) => write!(f, "{}", path.to_string_lossy()),
+        }
+    }
+}
+
+#[allow(dead_code)]
+pub struct OctezBakerNodeConfig {
```

Comment:
```suggestion
pub struct OctezBakerConfig {
```

Reply:
i've added node becuase it would conflict with `jstz/task/octez_baker::OctezBaker`,  can rename it back